src/storage/src/hummock/sstable/multi_builder.rs

```diff
@@ -264,15 +260,31 @@ where
     pub fn check_table_and_vnode_change(&mut self, user_key: &UserKey<&[u8]>) -> (bool, bool) {
         let mut switch_builder = false;
         let mut vnode_changed = false;
-        if self.split_by_table && user_key.table_id.table_id != self.last_table_id {
-            // table_id change
-            self.last_table_id = user_key.table_id.table_id;
-            switch_builder = true;
-            self.last_vnode = 0;
-            vnode_changed = true;
-            if self.split_weight_by_vnode > 1 {
-                self.largest_vnode_in_current_partition =
-                    VirtualNode::COUNT / (self.split_weight_by_vnode as usize) - 1;
+        if user_key.table_id.table_id != self.last_table_id {
+            let new_vnode_partition_count =
+                self.table_partition_vnode.get(&user_key.table_id.table_id);
+
+            if new_vnode_partition_count.is_some()
+                || self.table_partition_vnode.contains_key(&self.last_table_id)
+            {
+                if new_vnode_partition_count.is_some() {
+                    self.split_weight_by_vnode = *new_vnode_partition_count.unwrap();
+                } else {
+                    self.split_weight_by_vnode = 0;
+                }
+
+                // table_id change
+                self.last_table_id = user_key.table_id.table_id;
+                switch_builder = true;
+                self.last_vnode = 0;
+                vnode_changed = true;
+                if self.split_weight_by_vnode > 1 {
+                    self.largest_vnode_in_current_partition =
+                        VirtualNode::COUNT / (self.split_weight_by_vnode as usize) - 1;
+                } else {
+                    // default
+                    self.largest_vnode_in_current_partition = VirtualNode::MAX.to_index();
```

Comment:
Did we forget to set largest_vnode_in_current_partition when split_weight_by_vnode is 1 before this PR? Will it cause problem?

Reply:
```
largest_vnode_in_current_partition: VirtualNode::MAX.to_index()
```

the default value of largest_vnode_in_current_partition is `VirtualNode::MAX.to_index()`, it wil not cause problem even if we set the split_weight_by_vnode to `0` or `1` .

In short, we can turn off this feature by setting vnode_partition_count to 0